ipamapi/contract.go

```diff
@@ -50,6 +50,7 @@ var (
 	ErrIPOutOfRange        = types.BadRequestErrorf("Requested address is out of range")
 	ErrPoolOverlap         = types.ForbiddenErrorf("Pool overlaps with other one on this address space")
 	ErrBadPool             = types.BadRequestErrorf("Address space does not contain specified address pool")
+	ErrNoIPReturned        = types.NoServiceErrorf("No address returned")
```

Comment:
Looking at this new error, it makes me think we may want to follow the same idea but for the Pool, something like
`ErrNoPoolReturned`.
Once a a pool is not returned for a network, libnetwork would not even query the driver for an address, during create endpoint.